### ISSUE TITLE
Reason for function `prepare_data` changing order in batch based on caption length?

### ISSUE THREAD

**Issue by rubencart** [tohinz/semantic-object-accuracy-for-generative-text-to-image-synthesis#14]
As the title says, what's the reason that the `prepare_data` function in `datasets.py` sorts the batch elements based on caption length?

**Comment by tohinz** [comment]
Hi, that's because the text is processed by a RNN (LSTM or GRU) to obtain the text embedding. For this we use [pack_padded_sequence](https://pytorch.org/docs/stable/generated/torch.nn.utils.rnn.pack_padded_sequence.html) as preprocessing and `batch_first=True` (see [here](https://github.com/tohinz/semantic-object-accuracy-for-generative-text-to-image-synthesis/blob/bd00ed59fe6e3621b6baa977b74838c6b09d7d62/OP-GAN/code/model.py#L213)). This is standard procedure for RNNs in PyTorch to speed up training etc

**Comment by rubencart** [comment]
Okay I see thanks :). But shouldn't you change the bbox order as well then? ([here](https://github.com/tohinz/semantic-object-accuracy-for-generative-text-to-image-synthesis/blob/bd00ed59fe6e3621b6baa977b74838c6b09d7d62/OP-GAN/code/datasets.py#L22))

**Comment by tohinz** [comment]
I only used the bboxes during evaluation for visualization purposes (and yes, then they would have to be sorted which I forgot to do here, otherwise the wrong bboxes will be drawn on the generate image).
But generally, the network does not use the bboxes directly but uses the transformation matrices which are calculated from the bboxes [here](https://github.com/tohinz/semantic-object-accuracy-for-generative-text-to-image-synthesis/blob/bd00ed59fe6e3621b6baa977b74838c6b09d7d62/OP-GAN/code/datasets.py#L325). And the transformation matrices are indeed sorted ([see here](https://github.com/tohinz/semantic-object-accuracy-for-generative-text-to-image-synthesis/blob/bd00ed59fe6e3621b6baa977b74838c6b09d7d62/OP-GAN/code/datasets.py#L38))
